lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosAlgorithm.cxx

```diff
@@ -150,48 +150,57 @@ FunctionalChaosAlgorithm::FunctionalChaosAlgorithm(const Sample & inputSample,
 
 /* Constructor */
 FunctionalChaosAlgorithm::FunctionalChaosAlgorithm(const Sample & inputSample,
-    const Sample & outputSample)
-  : MetaModelAlgorithm(Distribution(), DatabaseFunction(inputSample, outputSample))
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : MetaModelAlgorithm(distribution, DatabaseFunction(inputSample, outputSample))
   , adaptiveStrategy_()
   , projectionStrategy_()
-  , maximumResidual_(ResourceMap::GetAsScalar( "FunctionalChaosAlgorithm-DefaultMaximumResidual" ))
+  , maximumResidual_(ResourceMap::GetAsScalar("FunctionalChaosAlgorithm-DefaultMaximumResidual"))
 {
   // Check sample size
   if (inputSample.getSize() != outputSample.getSize()) throw InvalidArgumentException(HERE) << "Error: the input sample and the output sample must have the same size.";
-  // Recover the distribution
-  LOGINFO("In FunctionalChaosAlgorithm, identify marginal distribution");
-  setDistribution(BuildDistribution(inputSample));
+
   const UnsignedInteger inputDimension = inputSample.getDimension();
   Collection< OrthogonalUniVariatePolynomialFamily > polynomials(inputDimension);
   for (UnsignedInteger i = 0; i < inputDimension; ++ i)
-  {
     polynomials[i] = StandardDistributionPolynomialFactory(getDistribution().getMarginal(i));
-  }
 
-  const HyperbolicAnisotropicEnumerateFunction enumerate(inputDimension, ResourceMap::GetAsScalar( "FunctionalChaosAlgorithm-QNorm" ));
+  const HyperbolicAnisotropicEnumerateFunction enumerate(inputDimension, ResourceMap::GetAsScalar("FunctionalChaosAlgorithm-QNorm"));
   OrthogonalProductPolynomialFactory basis(polynomials, enumerate);
-  const UnsignedInteger maximumTotalDegree = ResourceMap::GetAsUnsignedInteger( "FunctionalChaosAlgorithm-MaximumTotalDegree" );
   // For small sample size, use sparse regression
   LOGINFO("In FunctionalChaosAlgorithm, select adaptive strategy");
-  if (inputSample.getSize() < ResourceMap::GetAsUnsignedInteger( "FunctionalChaosAlgorithm-SmallSampleSize" ))
```

Comment:
Why don't we update resource map keys as we allow `string` ones?
We can have :
 - `FunctionalChaosAlgorithm-UseSparse` : boolean 
 - `FunctionalChaosAlgorithm-FittingAlgorithm` : string with possible values (`kfold`, `cloo`)

It is not easy to navigate and parametrise the choice. We possibly need to set `3 keys` (if `sample size` < `SmapllSampleSize`  to have a `full` chaos depending on data size whereas with this kind of proposition, only `1` is needed

Reply:
what do you think @regislebrun, ok to ditch the old logic ?
what would be the default then ? the cheapest  => usesparse=false ?

Reply:
ok,ditching the old Large/SmallSize keys, new keys:
FunctionalChaosAlgorithm-Sparse=False
FunctionalChaosAlgorithm-FittingAlgorithm=CorrectedLeaveOneOut